File: Data-Structures/Stacks/stacks.java
// Java Implementation of a Stack
class Stacks {
    private int arr[];
    private int top;
    private int capacity;

    public Stack(int size) {
        arr = new int[size];
        capacity = size;
        top = -1;
    }

    public void push(int value) {
        if (top >= capacity - 1) {
            System.out.println("Stack Overflow!");
            return;
        }
        arr[++top] = value;
        System.out.println("Pushed: " + value);
    }

    public void pop() {
        if (top < 0) {
            System.out.println("Stack Underflow!");
            return;
        }
        System.out.println("Popped: " + arr[top--]);
    }

    public void peek() {
        if (top < 0) {
            System.out.println("Stack is empty!");
            return;
        }
        System.out.println("Top Element: " + arr[top]);
    }

    public boolean isEmpty() {
        return top < 0;
    }

    public static void main(String[] args) {
        Stack s = new Stack(100);
        s.push(10);
        s.push(20);
        s.peek();
        s.pop();
        s.peek();
    }
}

File: Data-Structures/Stacks/stacks.cpp
// C++ Implementation of a Stack
#include <iostream>
using namespace std;

class Stack {
private:
    int arr[100];
    int top;
public:
    Stack() { top = -1; }
    
    void push(int value) {
        if (top >= 99) {
            cout << "Stack Overflow!" << endl;
            return;
        }
        arr[++top] = value;
        cout << "Pushed: " << value << endl;
    }
    
    void pop() {
        if (top < 0) {
            cout << "Stack Underflow!" << endl;
            return;
        }
        cout << "Popped: " << arr[top--] << endl;
    }
    
    void peek() {
        if (top < 0) {
            cout << "Stack is empty!" << endl;
            return;
        }
        cout << "Top Element: " << arr[top] << endl;
    }
    
    bool isEmpty() {
        return top < 0;
    }
};

// Example Usage
int main() {
    Stack s;
    s.push(10);
    s.push(20);
    s.peek();
    s.pop();
    s.peek();
    return 0;
}
